### DISCUSSION TITLE
Script closes after extracting dependencies

### DISCUSSION THREAD

**Discussion by Hopeepoh** [illsk1lls/ZipRipper#56] [Q&A]
Closing after dependencies and pop up nothing also there is no new file created

**Comment by Roger556677** [comment]
x2

**Comment by illsk1lls** [comment]
How large are your archives?

**Reply by mehaseeburrehman** [reply]
same case and file size 5gb

**Reply by illsk1lls** [reply]
Its the file size, large hashes are crashing John I may implement a limit with a MSGbox im just not sure how high we can go. I'll do testing on the upper limit today and get it added, and also see if there is a way to handle the larger files

**Comment by illsk1lls** [comment]
A crash was resolved in the latest push. Let me know if it resolves your issue.

**Reply by Hi-FiMan** [reply]
I'm still having this problem. Either the window closes after immediately displaying something I can't read (too fast) or it just sits there forever with a blinking cursor after extracting dependencies. The file is only 2.66GB in size.

I was able to screenshot where exactly it is crashing. it says creating hash and almost immediately closes after that.

**Reply by illsk1lls** [reply]
what kind of file is it?

can you check (if it crashed it didnt clean up after itself) `C:\%programdata%\JtR\run` and look for 2 files, `pwhash` and `statusout` and make sure pwhash isnt 0 bytes, if it is empty post statusout's contents here

**Reply by Hi-FiMan** [reply]
It's a rar file. pwhash has a handful of lines and statusout is blank.

**Reply by illsk1lls** [reply]
how big is pwhash? also if youre using a GPU what kind is it.

if you browse to that folder in CMD prompt, and type `john pwhash` does it run?

**Reply by Hi-FiMan** [reply]
It's 1KB in size and has 8 hash entries. I'm using an GeForce RTX 4070 with driver 566.36. I've actually been running john directly for an hour or so already via opencl. It seems to be doing something as one thread and my GPU are maxed out.
![1](https://github.com/user-attachments/assets/8229002e-66f1-4038-83e8-552525b9c443)

**Reply by illsk1lls** [reply]
Maybe try deleting lines 541-546 in the script and see if it works, it might be that for loop.. I'm not asking you to stop the job for me though, whenever you get a chance lmk

**Reply by illsk1lls** [reply]
is it a multipart rar file? if so did you join it ahead of time?

**Reply by illsk1lls** [reply]
I made a few small quoting updates. I tried to test with known bad chars in the internal filenames and main archive name, I cant reproduce on my end atm..

**Reply by Hi-FiMan** [reply]
It is indeed a two part rar. Do I need to combine them first? If so, how do I do that in this case?

**Reply by Hi-FiMan** [reply]
OK, so I retried the script on the first part which is over 3GB and it doesn't even spit out hashes. So now, I ran rar2john directly on the first part as I noticed it has complete files that don't seem to spill over into part 2. That spit out a load of hashes (one per file?) and now I am running john on those hashes. We will see what happens. I also renamed the files by replacing the spaces with underscores but I haven't tried the script on those yet.

I was reading this: https://github.com/openwall/john/issues/4346
They don't talk about combining the rar; but do mention an issue with files being split across multiple parts. So, if I am reading this right, I need to run the hashes on the part that has some complete self contained files which in my case is part 1.

**Reply by illsk1lls** [reply]
Ok, for ZipRipper you need to combine the archive. Use the following command in the folder containing the pieces.

`copy /B Filename.*.rar Filename.rar`

And then work on the single file. Let me know if that helps.

**Reply by Hi-FiMan** [reply]
Is that necessary for john or rar2john? Is it just for the script?

**Reply by illsk1lls** [reply]
both. If youre using ZipRipper use it on the output file. If your using rar2john same thing. Don't use the pieces.

**Reply by illsk1lls** [reply]
I found a bug as well that is now fixed. Special chars on the inside filenames would cause the script to crash when it was processing the contents of the hash.  It does this to determine if it can enable OpenCL and what type to use, e.g. RAR, RAR3, RAR5

**Reply by Hi-FiMan** [reply]
OK, I combined the files and the script still closes out but it did output hashes. I'm going to run john directly and see what happens.

**Reply by illsk1lls** [reply]
I did one last update. Not sure if it will help though. Make sure to refresh github with shift+reload.  Either way I'm glad you're able to proceed. I was able to squash one reproducible bug ;)

**Reply by Hi-FiMan** [reply]
Is it normal for john to sit there at this point?
![1](https://github.com/user-attachments/assets/d8d62d72-d704-46e3-9119-6c4e2bbb232a)

**Reply by illsk1lls** [reply]
yes it is working, press any key for status, 6 lines up from the bottom it shows some info
